### PR TITLE
Split configmaps and leases in different entries on vault-secret-operator role

### LINKED ISSUE #136: Issues applying the helm chart Role in namespace
In my current deployment scenario I am restricted to a cluster in which I am an admin in just a namespace. Current versions are:

```sh
kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:34:20Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.11", GitCommit:"27522a29febbcc4badac257763044d0d90c11abd", GitTreeState:"clean", BuildDate:"2021-09-15T19:16:25Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.22) and server (1.20) exceeds the supported minor version skew of +/-1
```

When I'm trying to apply the Role, I receive the following error with the following verbs:
```sh
{APIGroups:[""], Resources:["leases"], Verbs:["create" "delete" "get" "list" "patch" "update" "watch"]}
{APIGroups:["coordination.k8s.io"], Resources:["configmaps"], Verbs:["create" "delete" "get" "list" "patch" "update" "watch"]}
```

I arrived at the conclusion that the cluster is trying to authorize the target with some API groups that don't match 
ex. configmaps in "coordination.k8s.io".

Found the following problem lines 
https://github.com/ricoberger/vault-secrets-operator/blob/3d79175cc5988395d7b3a13550965e87d2e39107/charts/vault-secrets-operator/templates/role.yaml#L10-L23

Where 2 api groups are used at the same time.

After templating the chart, separating the API role descriptions and applying, everything seems to work as intended.

### PR DESCRIPTION
**Pull request by ecojan**
Fixes #136 
Changes done:
- Split configmaps and leases in different entries on vault-secret-operator role

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -7,10 +7,20 @@ metadata:
   name: vault-secrets-operator
 rules:
 - apiGroups:
-  - coordination.k8s.io
   - ""
   resources:
-  - configmaps
+  - leases
```
Comment (ricoberger) [inline config/rbac/role.yaml:12]: Shouldn't this be `configmaps` instead of `leases` like in the other file?

**Comment by ricoberger** [comment]
Hi @ecojan, thanks for your contribution 🙂.

Can you please check my comment before we merge this.

File: config/rbac/role.yaml (hunk 1 above)
Comment (ecojan) [reply config/rbac/role.yaml]: Done! Thank you for catching this!

**Comment by ecojan** [comment]
Thank you @ricoberger

**Comment by ricoberger** [comment]
Thanks again 🙂

**Pull request merged**